### PR TITLE
Created the new event dialog

### PR DESCRIPTION
**Pull request by garzeah**
### What this PR does (required):
- Created a modal for creating events
- Enabled the use of snackbars for error handling

### Screenshots / Videos (required):
Screen Shot 2021-07-11 at 11 52 39 PM

### Any information needed to test this feature (required):
- Head to the dashboard and  click the "New Event Type" button

### CODE REVIEW

**Comment by moffatethan** [comment]
What are your thoughts on refactoring the new event UI to not be a text field but instead we could offer some common options like "30 minute", "60 minute", "120 minutes", etc. Then offer the option to do a custom select option which will just generate minutes in maybe 15 minute increments.

For example, 15 minutes -> 30 minutes -> etc. And they can choose any amount that way. Just wanted to throw this out there for discussion 😄

File: client/src/App.tsx [hunk 1]
```diff
@@ -15,24 +16,24 @@ function App(): JSX.Element {
...
-            <Redirect to="/login" />
-          </Route>
-        </Switch>
-        {/* </AuthProvider>
-        </SnackBarProvider> */}
+        <SnackBarProvider>
+          {/* <AuthProvider> */}
+          <Switch>
+            {/* <Route exact path="/login" component={Login} />
+            <Route exact path="/signup" component={Signup} /> */}
```
Comment (moffatethan) [inline client/src/App.tsx]: Could this be uncommented or removed?

**Review by moffatethan — APPROVED** [review]
Left a comment for discussion but otherwise, this PR is good.

**Comment by garzeah** [comment]
> What are your thoughts on refactoring the new event UI to not be a text field but instead we could offer some common options like "30 minute", "60 minute", "120 minutes", etc. Then offer the option to do a custom select option which will just generate minutes in maybe 15 minute increments.
> 
> For example, 15 minutes -> 30 minutes -> etc. And they can choose any amount that way. Just wanted to throw this out there for discussion 😄

Hey Ethan, I really like that idea! I think it looks nicer from a UI/UX perspective too. Here's what it looks like right now:

Screen Shot 2021-07-12 at 4 29 52 PM

Although, I haven't added the custom option but it is something I will look into if we finish the priority tasks on time.

**Pull request merged**